tensorflow/compiler/jit/xla_compilation_cache.cc

```diff
@@ -332,15 +436,23 @@ Status XlaCompilationCache::CompileImpl(
   DCHECK_NE(out_executable, nullptr);
   VLOG(2) << "XlaCompilationCache::Compile " << DebugString();
 
-  if (VLOG_IS_ON(2)) {
-    VLOG(2) << "num_inputs=" << args.size();
-    for (int i = 0, end = args.size(); i < end; i++) {
+  VLOG(2) << "num_inputs=" << args.size();
+  if (VLOG_IS_ON(3)) {
+    for (int i = 0; i < args.size(); i++) {
       VLOG(3) << i << ": " << args[i].HumanString();
     }
   }
+  absl::optional<int64> compile_threshold;
+  if (compile_mode == CompileMode::kLazy) {
+    compile_threshold = kDefaultCompilationThreshold;
+  } else if (compile_mode == CompileMode::kAsync) {
+    compile_threshold = 0; // For now, always compile right away.
+  }
 
   TF_ASSIGN_OR_RETURN(Signature signature, BuildSignature(function, args));
-  VLOG(2) << "Signature: " << signature.HumanString();
+  string function_name = function.name();
+  string human_signature = VLOG_IS_ON(3) ? signature.HumanString() : function_name;
+  VLOG(2) << "Signature: " << human_signature;
```

Comment:
Let's put all of this under `VLOG_IS_ON(2)` to avoid unnecessary copies.

Reply:
done